crates/bevy_ecs/src/entity/mod.rs

```diff
@@ -676,7 +679,9 @@ impl EntityMeta {
         generation: 0,
         location: EntityLocation {
             archetype_id: ArchetypeId::INVALID,
-            index: usize::MAX, // dummy value, to be filled in
+            archetype_index: u32::MAX, // dummy value, to be filled in
```

Comment:
This sort of thing makes me wonder if we should be newtyping these two u32 fields as well. Swapping them is really nasty, and it would make the code easier to follow.

Reply:
Probably a good idea. Ideally we should be able to newtype a bunch of the indicies as we have with IDs for both type safety and also to ensure users aren't doing row level arithemtic on them. This is a bit out of the scope of this PR though.

Reply:
This was done in #4878.